server/clip_server/model/model.py

```diff
@@ -858,11 +544,15 @@ def patch_float(module):
 def load_openclip_model(
     model_name: str,
     model_path: str,
-    device: torch.device = torch.device('cpu'),
+    device: Union[str, torch.device] = 'cpu',
     jit: bool = False,
+    precision: str = None,
     force_quick_gelu: bool = False,
+    force_custom_text: bool = False,
     pretrained_image: bool = False,
 ):
+    if precision is None:
+        precision = 'fp32' if device in ('cpu', torch.device('cpu')) else 'fp16'
```

Comment:
Please clean up the `precision` and `casting` stuff. It's too complicated.

Reply:
- openclip uses `Layernorm32` when precision is set `fp32`, use `Layernorm` when `fp16`.
- based on `cast dtype`, if is `fp32`, flash attn cannot be used, this is a bug.